marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/artifacts/clearUserArtifacts.sjs

```diff
@@ -0,0 +1,33 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+//No specific privilege available for writing step definition or legacy mappings
+xdmp.securityAssert("http://marklogic.com/data-hub/privileges/write-ingestion", "execute");
```

Comment:
I actually don't think we need any privilege checks here. The delete calls will blow up if the user doesn't have update permission on any user artifacts.